### PR TITLE
doc: releasenotes 2.5: Add CAN driver release notes

### PR DESCRIPTION
**Pull request by alexanderwachter**
This commit adds the release notes for the 2.5 release for the CAN driver.

### CODE REVIEW

File: doc/releases/release-notes-2.5.rst [hunk 1]
```diff
@@ -196,6 +196,27 @@ Drivers and Sensors
 
 * CAN
 
+  * We reworked the configuration API.
+    A user can now either specify the timing manually
+    (define prop segment, phase segment1, phase segment2 and prescaler),
+    or use a newly introduced algorithm to calculate optimal timing values from
+    a bitrate and sampling point. The bitrate and sample point can be specifier
```
Comment (martinjaeger) [inline doc/releases/release-notes-2.5.rst:203]:
```suggestion
    a bitrate and sample point. The bitrate and sample point can be specified
```

File: doc/releases/release-notes-2.5.rst [hunk 2]
```diff
@@ -196,6 +196,27 @@ Drivers and Sensors
...
+    in the device tree too. It is possible to change the timing values
+    at runtime now.
+
+  * zcan_frame struct was reworked due to undefined behavior.
+    The std_id (11-bit) and ext_id (29-bit) are merged to a single id
+    field (29-bit). The union of both IDs was removed.
+
+  * We made the CANbus API was made CAN-FD compatible.
```
Comment (martinjaeger) [inline doc/releases/release-notes-2.5.rst:211]:
```suggestion
  * The CAN bus API was made CAN-FD compatible.
```

**Review by martinjaeger — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-2.5.rst (hunk 2 above)
Comment (alexanderwachter) [reply doc/releases/release-notes-2.5.rst:210]: Passive voice should be avoided.

**Review by martinjaeger — APPROVED** [review]

File: doc/releases/release-notes-2.5.rst (hunk 2 above)
Comment (karstenkoenig) [reply doc/releases/release-notes-2.5.rst]: Really? All the other entries are passive and I also felt it was a bit personal for a changelog. Your call though, the content itself isn't affected :-)

**Review by karstenkoenig — APPROVED** [review]

File: doc/releases/release-notes-2.5.rst (hunk 2 above)
Comment (alexanderwachter) [reply doc/releases/release-notes-2.5.rst]: At least Grammarly told me that after I pasted it there :-)
Don't you think it's better to read? The passive voice feels boring, IMO.

**Pull request merged**